### PR TITLE
#263 - TrimmedDocker introduced

### PR DESCRIPTION
**Pull request by olenagerasimova**
For #263 I've introduced `Docker` implementation to cut prefixes from repo name.

### CODE REVIEW

**Comment by g4s8** [comment]
@olenagerasimova @olegmoz just to remember: Docker image may contain slash chars in the middle, e.g. user `user` in repo `repo` may have these images:
 -  `central.artipie.com/user/repo/foo/bar/baz:latest`
 - `central.artipie.com/user/repo/baz:latest`

And it will be different images.

File: src/test/java/com/artipie/docker/http/TrimmedDockerTest.java [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+import org.hamcrest.core.IsEqual;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+/**
+ * Test for {@link TrimmedDocker}.
+ * @since 0.4
+ */
+@SuppressWarnings("PMD.AvoidDuplicateLiterals")
+class TrimmedDockerTest {
```
Comment (olegmoz) [inline src/test/java/com/artipie/docker/http/TrimmedDockerTest.java:42]: @olenagerasimova what do you think about using `@ParameterizedTest` here? There are multiple tests with same code, but different input strings

File: src/main/java/com/artipie/docker/http/TrimmedDocker.java [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+    /**
+     * Ctor.
+     * @param origin Docker origin
+     * @param prefix Prefix to cut
+     */
+    public TrimmedDocker(final Docker origin, final String prefix) {
+        this(
+            origin,
+            Pattern.compile(String.format("^/?(?:%s\\/?)(.*)", prefix))
```
[the remark this answers is not in the document]
Comment (olenagerasimova) [reply src/main/java/com/artipie/docker/http/TrimmedDocker.java:56]: @olegmoz let's clarify: 
- given prefixes will not start/end with `/` and I should not worry about it here 
- regex could be simplified to `(?:%s)\\/(.+)` where `%s` is prefix 

correct?

[the remark this answers is not in the document]
Comment (olegmoz) [reply src/main/java/com/artipie/docker/http/TrimmedDocker.java:56]: @olenagerasimova yes, this is correct expression

**Comment by olenagerasimova** [comment]
@olegmoz thanks, corrected, take a look please

**Pull request merged**